arbitrator/src/arbitrator.cpp

```diff
@@ -94,23 +103,33 @@ namespace arbitrator
 
     void Arbitrator::planning_state()
     {
+        ROS_INFO("Aribtrator beginning planning process!");
         ros::Time planning_process_start = ros::Time::now();
         cav_msgs::ManeuverPlan plan = planning_strategy_.generate_plan();
-        ros::Time plan_end_time = arbitrator_utils::get_plan_end_time(plan);
-        ros::Time plan_start_time = arbitrator_utils::get_plan_start_time(plan);
-        ros::Duration plan_duration = plan_end_time - plan_start_time;
-
-        if (plan_duration < min_plan_duration_) 
+        if (!plan.maneuvers.empty()) 
         {
-            ROS_WARN_STREAM("Arbitrator is unable to generate a plan with minimum plan duration requirement!");
-        } 
-        else 
+            ros::Time plan_end_time = arbitrator_utils::get_plan_end_time(plan);
+            ros::Time plan_start_time = arbitrator_utils::get_plan_start_time(plan);
+            ros::Duration plan_duration = plan_end_time - plan_start_time;
+
+            if (plan_duration < min_plan_duration_) 
+            {
+                ROS_WARN_STREAM("Arbitrator is unable to generate a plan with minimum plan duration requirement!");
+            } 
+            else 
+            {
+                ROS_INFO_STREAM("Arbitrator is publishing plan " << plan.maneuver_plan_id << " of duration " << plan_duration << " as current maneuver plan");
+            }
+            final_plan_pub_.publish(plan);
+        }
+        else
         {
-            ROS_INFO_STREAM("Arbitrator is publishing plan " << plan.maneuver_plan_id << " of duration " << plan_duration << " as current maneuver plan");
+            ROS_WARN("Arbitrator was unable to generate a plan!");
         }
-        final_plan_pub_.publish(plan);
 
         next_planning_process_start_ = planning_process_start + time_between_plans_;
+
+        sm_->submit_event(ArbitratorEvent::PLANNING_COMPLETE);
```

Comment:
What happens after you submit this PLANNING_COMPLETE event? I do not think we want state machine to transit to another state?

Reply:
After we submit `PLANNING_COMPLETE` the state machine transitions to `WAITING` state, where it stays for the remainder of the time until the next planning cycle. It'll sleep for that much time, the transition back to `PLANNING` state.

Reply:
Cool!